src/hooks/api/useToggleWish.ts

```diff
@@ -0,0 +1,53 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+
+type UseToggleWishParams = {
+  productId: number;
+};
+
+const useToggleWish = ({ productId }: UseToggleWishParams) => {
+  const queryClient = useQueryClient();
+
+  const { mutate, isPending } = useMutation({
+    mutationFn: async (isWished: boolean) => {
+      return Promise.resolve({
+        isWished: !isWished,
+      });
+    },
+    onMutate: async isWished => {
+      await queryClient.cancelQueries({
+        queryKey: ["productsWish", productId],
+      });
+
+      const previous = queryClient.getQueryData<{
+        isWished: boolean;
+        wishCount: number;
+      }>(["productsWish", productId]);
+
+      queryClient.setQueryData(["productsWish", productId], {
+        isWished: !isWished,
+        wishCount: (previous?.wishCount ?? 0) + (isWished ? -1 : 1),
+      });
+
+      return { previous };
+    },
```

Comment:
```suggestion
    mutationFn: async (isWished: boolean) => {
      await queryClient.cancelQueries({
        queryKey: ["productsWish", productId],
      });

      const previous = queryClient.getQueryData<{
        isWished: boolean;
        wishCount: number;
      }>(["productsWish", productId]);

      queryClient.setQueryData(["productsWish", productId], {
        isWished: !isWished,
        wishCount: (previous?.wishCount ?? 0) + (isWished ? -1 : 1),
      });

      return { previous };
    },
```
이렇게 쓰여야 하지 않을까요..!?
onMutate로 분리하신 이유가 있을까요?

Reply:
mutationFn은 실제 서버 요청을 작성하는 부분 onMutate는 mutate가 실행되었을 때 서버 요청 앞에 적용되는 로직을 작성하는 것이라고 생각하고 있었습니다. 
현재 wish 버튼을 눌렀을 때 서버에 api 요청을 보내지 않아 Promise를 받고 onMutate에서 낙관적 업데이트를 적용하도록 구성했습니다. 
mutationFn에 서버 요청이 아닌 부분을 작성해도 괜찮나요?